### PR TITLE
feat: update kubo to v0.19

### PR DESCRIPTION
**Pull request by gfanton**
update kubo to v0.19.0

depends on:
- [x] https://github.com/berty/go-ipfs-log/pull/84
- [x] https://github.com/berty/go-orbit-db/pull/132
- [x] https://github.com/berty/weshnet/pull/57
- [x] https://github.com/berty/go-ipfs-repo-encrypted/pull/56

### CODE REVIEW

File: go/cmd/rdvp/main.go [hunk 1]
```diff
@@ -192,17 +189,15 @@ func main() {
 			defer host.Close()
 			logHostInfo(logger, host)
 
-			ressources := libp2p_relayv1.DefaultResources()
-			_, err = libp2p_relayv1.NewRelay(host,
-				libp2p_relayv1.WithResources(ressources),
-				// libp2p_relayv1.WithACL(acl)), // not use for now
+			_, err = libp2p_relayv2.New(host,
+				// disable limits for now to have an equivalent off a relay v1
```
Comment (jefft0) [inline go/cmd/rdvp/main.go:193]: Should be "equivalent of"

File: go/internal/initutil/ipfs.go [hunk 2]
```diff
@@ -495,11 +494,13 @@ func (m *Manager) setupIPFSConfig(cfg *ipfs_cfg.Config) ([]libp2p.Option, error)
...
 
 	// disable original connection manager
-	cfg.Swarm.ConnMgr.Type = "none"
+	cfg.Swarm.ConnMgr.Type = ipfs_cfg.NewOptionalString("none")
 
 	// configure custom conn manager
 	cm, err := connmgr.NewConnManager(
-		cfg.Swarm.ConnMgr.LowWater, cfg.Swarm.ConnMgr.HighWater, connmgr.WithGracePeriod(ipfs_cfg.DefaultConnMgrGracePeriod),
+		int(cfg.Swarm.ConnMgr.LowWater.WithDefault(50)),
+		int(cfg.Swarm.ConnMgr.HighWater.WithDefault(100)),
```
Comment (jefft0) [inline go/internal/initutil/ipfs.go:502]: Question: This is the code that should prevent the "too many open files error". Correct?

File: go/internal/initutil/ipfs.go [hunk 3]
```diff
@@ -589,8 +589,8 @@ func (m *Manager) setupIPFSConfig(cfg *ipfs_cfg.Config) ([]libp2p.Option, error)
 	// enable mdns
 	cfg.Discovery.MDNS.Enabled = false
 
-	// disable managed auto relay
-	cfg.Swarm.RelayClient.Enabled = ipfs_cfg.False
+	// make sure  auto relay is enable
```
Comment (D4ryl00) [inline go/internal/initutil/ipfs.go:592]: Why do you change autorelay to true?

File: go/internal/initutil/ipfs.go [hunk 4]
```diff
@@ -620,8 +617,8 @@ func (m *Manager) setupIPFSConfig(cfg *ipfs_cfg.Config) ([]libp2p.Option, error)
 
 	// @NOTE(gfanton): disable quic transport so we can init a custom transport
 	// with reusport disable
-	cfg.Swarm.Transports.Network.QUIC = ipfs_cfg.False
-	p2popts = append(p2popts, libp2p.Transport(quict.NewTransport, quict.DisableReuseport()))
+	cfg.Swarm.Transports.Network.QUIC = ipfs_cfg.True
```
Comment (D4ryl00) [inline go/internal/initutil/ipfs.go:620]: Is it ok the enable QUIC? Is reusePort disabled?

**Review by D4ryl00 — APPROVED** [review]
LGTM

File: go/internal/initutil/ipfs.go (hunk 4 above)
Comment (gfanton) [reply go/internal/initutil/ipfs.go:620]: that was a mistake, thanks !

File: go/internal/initutil/ipfs.go (hunk 3 above)
Comment (gfanton) [reply go/internal/initutil/ipfs.go:592]: that's a typo error, thanks !

**Pull request merged**